Fix issue #14877 - mouse wheel being intepreted as x-axis movement on linux

nick-l-o3de opened the pull request:
## What does this PR do?

Fixes the code in the XCB mouse input handler.
The code was previously assuming every raw motion event coming in was either for the x or y axis of the mouse.  However, raw motion events have a mask to describe which axis the event packet contains.  When you roll the wheel, or presumably, move the Z axis or other axes of a multi-axis device, it sets those bits to indicate which axes the data represents.

This change makes it so that the mouse movement code only considers the axes data if bit 0x1 or 0x2 are set (x and y mouse motion).
Wheel motion is captured by the button press code elsewhere.

## How was this PR tested?
Compile and building in linux debug and profile.  In debug, clear box testing and profile opaque box testing.  
The event now only triggers X and Y when I actuallymove the mouse X and y, and does not trigger if I roll the wheel anymore.

AMZN-alexpete commented on Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceMouse.cpp:
```diff
@@ -500,6 +500,35 @@ namespace AzFramework
+                // being the first two axes of the device, which for mice, is going to be the normal x and y axes.
+
+                // Anything beyond that is a completely different axis, such as the mouse wheel, z axis, other things
+                // special mice can do.  The
```
I'm not sure if maybe this was where you were going with this?
```suggestion
                // special mice can do.  The most common example of these are the mere protrusions into our dimension of vastly hyperintelligent pandimensional beings.
```
https://www.goodreads.com/quotes/475365-these-creatures-you-call-mice-you-see-they-are-not 😄  🐭

AMZN-alexpete reviewed (approved):
Thanks for the fix! Just a minor suggestion 🐭 👽

nick-l-o3de commented:
Oh, interesting. Some tests started to legit fail.  I'll look into this.

nick-l-o3de commented:
Ah, I see.  The test constructs a fake xcb event but doesn't fill in its mask value, which it should synthesize now that we're actually using the mask value...  I'll fix the test

nick-l-o3de commented:
I've fixed the tests by adding the new function that is called in xcb to the mock.
```
LIB: libAzFramework.Xcb.Tests.so
Loading: libAzFramework.Xcb.Tests.so
OKAY Library loaded: libAzFramework.Xcb.Tests.so
OKAY Symbol found: AzRunUnitTests
[==========] Running 10 tests from 4 test cases.
[----------] Global test environment set-up.
[----------] 2 tests from XcbInputDeviceKeyboardTests
[ RUN      ] XcbInputDeviceKeyboardTests.InputChannelsUpdateStateFromXcbEvents
[       OK ] XcbInputDeviceKeyboardTests.InputChannelsUpdateStateFromXcbEvents (5869 ms)
[ RUN      ] XcbInputDeviceKeyboardTests.TextEnteredFromXcbKeyPressEvents
[       OK ] XcbInputDeviceKeyboardTests.TextEnteredFromXcbKeyPressEvents (5828 ms)
[----------] 2 tests from XcbInputDeviceKeyboardTests (11697 ms total)

[----------] 2 tests from XcbInputDeviceMouseTests
[ RUN      ] XcbInputDeviceMouseTests.MovementInputChannelsUpdateStateFromXcbEvents
[       OK ] XcbInputDeviceMouseTests.MovementInputChannelsUpdateStateFromXcbEvents (5392 ms)
[ RUN      ] XcbInputDeviceMouseTests.MovementInputChannelsUpdateStateFromXcbEvents_MouseWheelDoesNotCountAsAxisMovement
[       OK ] XcbInputDeviceMouseTests.MovementInputChannelsUpdateStateFromXcbEvents_MouseWheelDoesNotCountAsAxisMovement (5371 ms)
[----------] 2 tests from XcbInputDeviceMouseTests (10763 ms total)

[----------] 3 tests from AllButtons/XcbInputDeviceMouseButtonTests
[ RUN      ] AllButtons/XcbInputDeviceMouseButtonTests.ButtonInputChannelsUpdateStateFromXcbEvents/0
[       OK ] AllButtons/XcbInputDeviceMouseButtonTests.ButtonInputChannelsUpdateStateFromXcbEvents/0 (5572 ms)
[ RUN      ] AllButtons/XcbInputDeviceMouseButtonTests.ButtonInputChannelsUpdateStateFromXcbEvents/1
[       OK ] AllButtons/XcbInputDeviceMouseButtonTests.ButtonInputChannelsUpdateStateFromXcbEvents/1 (5516 ms)
[ RUN      ] AllButtons/XcbInputDeviceMouseButtonTests.ButtonInputChannelsUpdateStateFromXcbEvents/2
[       OK ] AllButtons/XcbInputDeviceMouseButtonTests.ButtonInputChannelsUpdateStateFromXcbEvents/2 (5437 ms)
[----------] 3 tests from AllButtons/XcbInputDeviceMouseButtonTests (16525 ms total)

[----------] 3 tests from AllPointerPositions/XcbGetSystemCursorPositionTests
[ RUN      ] AllPointerPositions/XcbGetSystemCursorPositionTests.GetSystemCursorPositionNormalizedReturnsCorrectValue/0
[       OK ] AllPointerPositions/XcbGetSystemCursorPositionTests.GetSystemCursorPositionNormalizedReturnsCorrectValue/0 (5408 ms)
[ RUN      ] AllPointerPositions/XcbGetSystemCursorPositionTests.GetSystemCursorPositionNormalizedReturnsCorrectValue/1
[       OK ] AllPointerPositions/XcbGetSystemCursorPositionTests.GetSystemCursorPositionNormalizedReturnsCorrectValue/1 (5414 ms)
[ RUN      ] AllPointerPositions/XcbGetSystemCursorPositionTests.GetSystemCursorPositionNormalizedReturnsCorrectValue/2
[       OK ] AllPointerPositions/XcbGetSystemCursorPositionTests.GetSystemCursorPositionNormalizedReturnsCorrectValue/2 (5484 ms)
[----------] 3 tests from AllPointerPositions/XcbGetSystemCursorPositionTests (16306 ms total)

[----------] Global test environment tear-down
[==========] 10 tests from 4 test cases ran. (55291 ms total)
[  PASSED  ] 10 tests.
OKAY AzRunUnitTests() returned 0
Returning code: 0
```

I also added a new test for this regression.
